x-pack/platform/plugins/shared/onechat/server/services/tools/persisted/tool_types/index_search/to_tool_definition.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IndexSearchToolConfig } from '@kbn/onechat-common/tools';
+import { ToolType } from '@kbn/onechat-common';
+import { z } from '@kbn/zod';
+import { runSearchTool } from '@kbn/onechat-genai-utils/tools';
+import type { ToolPersistedDefinition } from '../../client';
+import type { InternalToolDefinition } from '../../../tool_provider';
+
+const searchSchema = z.object({
+  nlQuery: z.string().describe('A natural language query expressing the search request'),
+});
+
+type SearchSchemaType = typeof searchSchema;
+
+export function toToolDefinition(
+  tool: ToolPersistedDefinition<IndexSearchToolConfig>
+): InternalToolDefinition<IndexSearchToolConfig, SearchSchemaType> {
+  const { id, description, tags, configuration } = tool;
+  return {
+    id,
+    type: ToolType.index_search,
+    description,
+    tags,
+    configuration,
+    schema: searchSchema,
+    handler: async ({ nlQuery }, { esClient, modelProvider, logger }) => {
+      const { pattern } = configuration;
+      const results = await runSearchTool({
+        nlQuery,
+        index: pattern,
+        esClient: esClient.asCurrentUser,
+        model: await modelProvider.getDefaultModel(),
+        logger,
+      });
+      return { results };
+    },
```

Comment:
This is mostly the same code for all persisted tool types (except for the handler), so we may want to revisit later in some way, but for now it's fine.